docs/guides/environmental-variables.md

```diff
@@ -0,0 +1,57 @@
+---
+title: CLI
+---
+
+Environmental variables are key-value pairs that are stored in our operating systems. Many of them come by default, and we can also create our own. They are used to store information that is used by our operating system or by other programs. For example, the `PATH` variable stores a list of directories that the operating system should look in when we run a command in the terminal. The `HOME` variable stores the path to our home directory.
+
+The same can be used for building softwares aswell! Configuration values are often stored in environmental variables.
+
+## Need of Environmental Variables
+
+Applications use multiple external tools, authentication methods, and numerous configurations. Hence, for proper functioning, we need our code to use these values properly.
+
+Consider a simple scenario of JWT authentication. Typically, when signing the tokens for our users, we will need the following configuration set:
+
+- expiry time
+- secret key
+- issuer
+
+To go about this, there are broadly two ways:
+
+1. **Hardcode the values in our code** \
+   This is perhaps the easiest approach to solve the problem, but the most dangerous (and inefficient) one! Once you hardcode the values in your codebase, everyone who works on the code will be able to see your sensitive information, which poses a massive security issue. Also, if you want to change the values, you will have to change the code and redeploy the application, which is not ideal.
+
+2. **Store the values in environmental variables** \
+   This is the best approach to solve the problem. You can store the sensitive values in the OS of the server running your application, and during the runtime of your application, you can access these values from the OS. All programming languages have excellent support for this. This way, you can keep your sensitive information safe, and also change the values without having to redeploy the application.
```

Comment:
We will need to restart the tailcall server though.

Reply:
ahhh, yes, that's a case i missed.